action/parser.php

```diff
@@ -26,6 +27,39 @@ class action_plugin_prosemirror_parser extends DokuWiki_Action_Plugin
     public function register(Doku_Event_Handler $controller)
     {
         $controller->register_hook('ACTION_ACT_PREPROCESS', 'BEFORE', $this, 'handle_preprocess');
+        $controller->register_hook('COMMON_DRAFT_SAVE', 'BEFORE', $this, 'handle_draft');
+    }
+
+    /**
+     * Triggered by: COMMON_DRAFT_SAVE
+     *
+     * @param Doku_Event $event
+     * @param            $param
+     */
+    public function handle_draft(Doku_Event $event, $param)
+    {
+        global $INPUT, $MSG;
+        $unparsedJSON = $INPUT->post->str('prosemirror_json');
+        if (empty($unparsedJSON)) {
+            return;
+        }
+        $syntax = $this->getSyntaxFromProsemirrorData($unparsedJSON);
+
+        if (!empty($MSG) && $this->isAjax()) {
+            echo '<div class="js-prosemirror-draft-errors" style="display: none;">';
+            html_msgarea();
+            echo '</div>';
+        }
```

Comment:
This echos stuff back to the draft ajax call, right? This seems quite hacky. 
Could we throw an exception in getSyntaxFromProsemirrorData() instead and handle it properly?
Maybe we need a way to signal errors via the draft event?